pkgs/applications/networking/avalanchego/default.nix

```diff
@@ -0,0 +1,41 @@
+{ buildGoModule
+, fetchFromGitHub
+, lib
+, stdenv
+, IOKit
+}:
+
+buildGoModule rec {
+  pname = "avalanchego";
+  version = "1.9.5";
+
+  src = fetchFromGitHub {
+    owner = "ava-labs";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-Uim2Q3LvNEV0C5/gFdAiXKy2oKbhhc/ifRyz5BIh6zE=";
+  };
+
+  # go mod vendor has a bug, see: https://github.com/golang/go/issues/57529
+  proxyVendor = true;
+
+  subPackages = [ "main" ];
+
+  vendorHash = "sha256-43bUvEwuppizLybj32eolSdpnqGJkkdA53/uBZkCrDc=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/ava-labs/avalanchego/version.GitCommit=${version}"
+  ];
+
```

Comment:
```suggestion
# patch zondax/hid dependency manually on darwin until https://github.com/ava-labs/avalanchego/pull/1194 is merged
  postPatch = lib.optionalString stdenv.isDarwin ''
    substituteInPlace go.mod \
      --replace 'github.com/zondax/hid v0.9.1-0.20220302062450-5552068d2266' 'github.com/zondax/hid v0.9.1'
    substituteInPlace go.sum \
      --replace 'github.com/zondax/hid v0.9.0/go.mod h1:l5wttcP0jwtdLjqjMMWFVEE7d1zO0jvSPA9OPZxWpEM=' 'github.com/zondax/hid v0.9.1 h1:gQe66rtmyZ8VeGFcOpbuH3r7erYtNEAezCAYu8LdkJo=' \
      --replace 'github.com/zondax/hid v0.9.1-0.20220302062450-5552068d2266/go.mod h1:l5wttcP0jwtdLjqjMMWFVEE7d1zO0jvSPA9OPZxWpEM=' 'github.com/zondax/hid v0.9.1/go.mod h1:l5wttcP0jwtdLjqjMMWFVEE7d1zO0jvSPA9OPZxWpEM='
  '';
```

Reply:
Patching go.sum like this is bound to make go.mod unhappy. Please use a patch file.